src/gatsby/src/components/search/noSearchResults.js

```diff
@@ -2,93 +2,133 @@ import React from "react"
+    >
+      Map of parks
+      <FontAwesomeIcon icon={faArrowUpRightFromSquare} className="link-icon" />
+    </a>
+  )
+}
+
 const NoSearchResults = ({
-  hasCity,
-  hasPark,
-  hasFilter,
   handleClickClearCity,
-  handleKeyDownClearCity,
   handleClickClearPark,
-  handleKeyDownClearPark,
   handleClickClearFilter,
-  handleKeyDownClearFilter,
+  page,
 }) => {
+  const renderNoResultsMessage = () => {
+    switch (page) {
+      case "find-a-park":
+        return "No parks match your search."
+      case "park-operating-dates":
+      case "a-z-list":
+        return "No parks match your selection."
+      case "approved":
+        return "No documents match your selection."
+      default:
+        return ""
+    }
+  }
+  // event handlers
+  const handleClearAll = () => {
+    handleClickClearCity()
+    handleClickClearPark()
+    handleClickClearFilter()
+  }
+  const handleKeyDownClearAll = e => {
+    if (e.key === "Enter" || e.key === " ") {
+      e.preventDefault()
+      handleClearAll()
+    }
+  }
+
   return (
     <div className="no-results-container">
       <div className="no-results-header">
         <FontAwesomeIcon icon={faSearch} className="search-icon" />
         <div className="no-results-header--right">
           <h2>Sorry...</h2>
-          <p><b>no parks matched your search.</b></p>
+          <p><b>{renderNoResultsMessage()}</b></p>
         </div>
       </div>
-      <p><b>You could:</b></p>
-      <ul>
-        {(hasCity || hasPark) && (
-          <li>Change your search term</li>
-        )}
-        {hasCity && (
-          <li>
-            <button
-              tabIndex={0}
-              className="btn btn-link"
-              aria-label="Remove city name"
-              onClick={handleClickClearCity}
-              onKeyDown={handleKeyDownClearCity}
+      {page === "find-a-park" && (
+        <>
+          <p><b>You could:</b></p>
+          <ul>
+            <li>Try a different search term</li>
+            <li>
+              <button
+                tabIndex={0}
+                className="btn btn-link"
+                aria-label="Clear all"
+                onClick={handleClearAll}
+                onKeyDown={handleKeyDownClearAll}
+              >
+                Go back to the full list
+              </button>
+            </li>
+          </ul>
+          <p><b>Or search using the:</b></p>
+          <ul>
+            <li>
+              <Link to="/find-a-park/a-z-list">A–Z park list</Link>
+            </li>
+            <li>
+              <MapLink />
+            </li>
+          </ul>
+        </>
+      )}
+      {(page === "park-operating-dates" || page === "a-z-list") && (
+        <>
+          <p className="mb-4">
+            <b>Select 'All' to go back to the full list.</b>
+          </p>
+          <p><b>Or search using the:</b></p>
+          <ul>
+            <li>
+              <Link to="/find-a-park">Find a park</Link> page
```

Comment:
I remember we updated the links in #1531 to include "page" in the anchor text; do we want to do that here too? I see "...list" is already included in the A-Z list link.

Reply:
Thanks! Brett prefers to include "page" as a link.